src/transformers/models/maskformer/modeling_maskformer.py

```diff
@@ -0,0 +1,2502 @@
+        }
+        return weighted_loss_dict
+
+    def get_loss(self, loss_dict: Dict[str, Tensor]) -> Tensor:
+        return sum(loss_dict.values())
+
+    def get_logits(self, outputs: MaskFormerOutput) -> Tuple[Tensor, Tensor, Dict[str, Tensor]]:
+        pixel_embeddings = outputs.pixel_decoder_last_hidden_state
+        # get the auxilary predictions (one for each decoder's layer)
+        auxilary_logits: List[str, Tensor] = []
+        # This code is a little bit cumbersome, an improvement can be to return a list of predictions. If we have auxilary loss then we are going to return more than one element in the list
+        if self.config.use_auxilary_loss:
+            stacked_decoder_outputs = torch.stack(outputs.transformer_decoder_hidden_states)
+            classes = self.class_predictor(stacked_decoder_outputs)
+            class_queries_logits = classes[-1]
+            # get the masks
+            mask_embeddings = self.mask_embedder(stacked_decoder_outputs)
+            # sum up over the channels for each embedding
+            binaries_masks = torch.einsum("lbqc,   bchw -> lbqhw", mask_embeddings, pixel_embeddings)
+            masks_queries_logits = binaries_masks[-1]
+            # go til [:-1] because the last one is always used
+            for aux_binary_masks, aux_classes in zip(binaries_masks[:-1], classes[:-1]):
+                auxilary_logits.append({"masks_queries_logits": aux_binary_masks, "class_queries_logits": aux_classes})
+
+        else:
+            transformer_decoder_hidden_states = outputs.transformer_decoder_last_hidden_state
+            classes = self.class_predictor(transformer_decoder_hidden_states)
+            class_queries_logits = classes
+            # get the masks
+            mask_embeddings = self.mask_embedder(transformer_decoder_hidden_states)
+            # sum up over the channels
+            masks_queries_logits = torch.einsum("bqc,   bchw -> bqhw", mask_embeddings, pixel_embeddings)
+
+        return class_queries_logits, masks_queries_logits, auxilary_logits
+
+    @add_start_docstrings_to_model_forward(MASKFORMER_INPUTS_DOCSTRING)
+    @replace_return_docstrings(output_type=MaskFormerForInstanceSegmentationOutput, config_class=_CONFIG_FOR_DOC)
+    def forward(
+        self,
+        pixel_values: Tensor,
+        mask_labels: Optional[Tensor] = None,
+        class_labels: Optional[Tensor] = None,
+        pixel_mask: Optional[Tensor] = None,
+        output_hidden_states: Optional[bool] = None,
+        output_attentions: Optional[bool] = None,
+        return_dict: Optional[bool] = True,
+    ) -> MaskFormerForInstanceSegmentationOutput:
+        r"""
+        mask_labels (`torch.FloatTensor`, *optional*, defaults to `None`):
+            A tensor of shape `(num_classes, height, width)`.
+        class_labels (`torch.LongTensor`, *optional*, defaults to `None`)::
+            A tensor of shape (`num_classes).
```

Comment:
No defaults to None as seen before.
Also, the last shape is missing a `
```suggestion
        mask_labels (`torch.FloatTensor`, *optional*):
            A tensor of shape `(num_classes, height, width)`.
        class_labels (`torch.LongTensor`, *optional*):
            A tensor of shape (`num_classes`).
```

Finally there should be some documentation on what each label is.